src/UIHandler.h

```diff
@@ -0,0 +1,110 @@
+#pragma once
+#ifndef UIHANDLER_H
+#define UIHANDLER_H
+
+#include "debug.h"
+
+#include "cursesdef.h"
+#include "output.h"
+
+struct IntPair
+{
+    int x;
+    int y;
+};
+
+class UIPanel
+{
+public:
+    virtual std::vector<UIPanel*> getChild() = 0;
+    virtual void addChild(UIPanel *panel) = 0;
+    virtual void removeChild(size_t index) = 0; // Passes back removed child
+
+    virtual IntPair RequestedSize(bool min) = 0;
```

Comment:
I don't get the meaning of the parameter. `bool` parameters are always suspect. It's probably better to make this a private function and have *two* separate public function that call it:
```C++
IntPair RequestMinSize() const { return RequestSize(true); }
IntPair RequestMaxSize() const { return RequestSize(false); }
```

Even better would be an `enum`, because nobody will remember whether the bool means `min` or `max` (whether `RequestSize(true)` returns the minimal size or the maximal size). Or does it return the ***true*** and the ***untrue*** size, whatever that means?

```C++
enum Foo { // find better name
    MININAL, MAXIMAL
};
void RequestSize(Foo f);
```